src/consumption.cpp

```diff
@@ -638,19 +638,24 @@ bool player::eat( item &food, bool force )
                                  units::to_milliliter( stomach.contains() ) ) > units::to_milliliter(
                                 stomach.capacity() );
     const bool saprophage = has_trait( trait_id( "SAPROPHAGE" ) );
+    bool eaten = true;
```

Comment:
`consume_effects( food )` is called all paths of the if-else block, so you can just put it at the end of the block:

```CPP
    if( !consume_effects( food ) ) {
        return false;
    }
```

Reply:
Aha, good point.